### PR TITLE
Remove old agents page

### PR DESCRIPTION
**Pull request by iamsumanth**
* Environments page uses old agents table to edit agents, need to find alternate.

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/views/layouts/agent_detail.html.erb [hunk 1]
```diff
@@ -19,7 +19,6 @@
       <div class="page_header">
         <div class="row">
           <ul class="entity_title">
-            <!--<li class="name"><%= link_to l.string("AGENTS_POPUP_TITLE"), url_for(:controller => "agents", :action => "index") -%> </li>-->
```
Comment (jyotisingh) [inline server/webapp/WEB-INF/rails.new/app/views/layouts/agent_detail.html.erb]: why did you get rid of the link? wouldn't pointing to the new page suffice?

File: server/src/com/thoughtworks/go/server/service/AgentService.java [hunk 2]
```diff
@@ -114,8 +114,8 @@ public void sync(Agents agents) {
     }
 
     @Deprecated // used by old code, stop using AgentViewModel
-    public AgentsViewModel agents() {
-        return toAgentViewModels(agentInstances.allAgents());
+    public AgentInstances agents() {
```
Comment (jyotisingh) [inline server/src/com/thoughtworks/go/server/service/AgentService.java]: No one seems to be using this method apart from the old controller which is being deleted as a part of this PR. Why not just get rid of the method?

File: server/test/integration/com/thoughtworks/go/server/service/AgentServiceIntegrationTest.java [hunk 3]
```diff
@@ -255,22 +254,6 @@ public void shouldRemoveResourcesFromMultipleAgents() {
     }
 
     @Test
-    public void shouldFindAnAgentForAGivenUUID() {
```
Comment (jyotisingh) [inline server/test/integration/com/thoughtworks/go/server/service/AgentServiceIntegrationTest.java]: `findAgentAndRefreshStatus` still exists in AgentService. Why delete the test?

File: server/webapp/WEB-INF/rails.new/app/views/layouts/agent_detail.html.erb (hunk 1 above)
Comment (iamsumanth) [inline server/webapp/WEB-INF/rails.new/app/views/layouts/agent_detail.html.erb]: Link is already commented and we are not using it anymore to point to agents page. Thats why I removed it.

**Pull request merged**